### PR TITLE
rewrite_comment: fix block fallback when failing to rewrite an itemized block

### LINKED ISSUE #3224: Regression: wrap_comments with a markdown list
I have this list in a module doc comment:

```rust
//! * [`examples/simple`] – Demonstrates use of the [`init`] API with plain
//!   structs.
//! * [`examples/simple_flatbuffer`] – Demonstrates use of the [`init`] API with
//!   FlatBuffers.
//! * [`examples/gravity`] – Demonstrates use of the [`RLBot::set_game_state`]
//!   API
```

Recently, this diff appeared in CI:

```diff
 //! * [`examples/simple`] – Demonstrates use of the [`init`] API with plain
 //!   structs.
 //! * [`examples/simple_flatbuffer`] – Demonstrates use of the [`init`] API with
-//!   FlatBuffers.
+//! FlatBuffers.
 //! * [`examples/gravity`] – Demonstrates use of the [`RLBot::set_game_state`]
 //!   API
```

For some reason rustfmt decided that the middle item, and only the middle item, should not indent past the bullet point.

Unfortunately I don't work on this crate too often so I don't have a good idea of when it regressed. The last successful build used the nightly from Nov 3. The failure used today's nightly.

### PR DESCRIPTION
**Pull request by scampi**
close #3224

### CODE REVIEW

**Comment by scampi** [comment]
@nrc @topecongiro is there a change needed for this PR ?

File: src/comment.rs [hunk 1]
```diff
@@ -456,10 +459,32 @@ impl ItemizedBlock {
...
+            .map(|line| format!("{} ", line.trim_start()))
+            .collect::<String>()
+    }
+
+    /// Returns the block as a string under its original form
+    fn original_block_as_string(&self) -> String {
+        self.lines
+            .iter()
+            .map(|line| format!("{}\n", line))
+            .collect::<String>()
```
Comment (topecongiro) [inline src/comment.rs]: We could simplify this chain as `self.lines.join("\n")`.

**Comment by scampi** [comment]
thanks for the review @topecongiro I forgot to commit the test case...

**Comment by scampi** [comment]
@topecongiro is it possible to re-run the tests ? They failed with `error: failed to install component: 'rust-std-x86_64-unknown-linux-gnu', detected conflict: '"lib/rustlib/x86_64-unknown-linux-gnu/lib/libLLVM-8svn.so"'` which I think has been resolved a while back.

**Comment by nrc** [comment]
I restarted, but you might need to rebase to get the fix in order to pass CI

**Comment by otavio** [comment]
This seems good to go as CI is happy now ;-)

**Pull request merged**